sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py

```diff
@@ -69,14 +69,17 @@ def list(self, *, compute_type: Optional[str] = None) -> Iterable[Compute]:
                 :caption: Retrieving a list of the AzureML Kubernetes compute resources in a workspace.
         """
 
-        return self._operation.list(
-            self._operation_scope.resource_group_name,
-            self._workspace_name,
-            cls=lambda objs: [
-                Compute._from_rest_object(obj)
-                for obj in objs
-                if compute_type is None or Compute._from_rest_object(obj).type.lower() == compute_type.lower()
-            ],
+        return cast(
```

Comment:
No cast

Reply:
`self.operation.list` function return `Iterable[PaginatedComputeResourcesList]` type, but here it is expecting to return `Iterable[Compute]` type:

![image](https://github.com/Azure/azure-sdk-for-python/assets/19650434/01a17221-e806-4dac-8010-630f38f75ab4)

Reply:
Let's use a type: ignore here. The `cls` param is already handling the casting with the lambda function. MyPy just isn't smart enough to notice.

Reply:
Fixed